packages/common-library/src/common_library/error_codes.py

```diff
@@ -22,9 +24,24 @@
         str, StringConstraints(strip_whitespace=True, pattern=_PATTERN)
     ]
 
+_LEN = 12  # chars (~48 bits)
+
+
+def _generate_error_fingerprint(exc: BaseException) -> str:
+    """
+    Unique error fingerprint for deduplication purposes
+    """
+    tb = traceback.extract_tb(exc.__traceback__)
+    frame_sigs = [f"{frame.name}:{frame.lineno}" for frame in tb]
+    fingerprint = f"{type(exc).__name__}|" + "|".join(frame_sigs)
+    # E.g. ZeroDivisionError|foo:23|main:10
+    return hashlib.sha256(fingerprint.encode()).hexdigest()[:_LEN]
+
 
 def create_error_code(exception: BaseException) -> ErrorCodeStr:
-    return TypeAdapter(ErrorCodeStr).validate_python(_LABEL.format(id(exception)))
+    return TypeAdapter(ErrorCodeStr).validate_python(
```

Comment:
While this is nice. I would still like to have the ability to track unique errors. Could we maybe add a suffix on the error code so that the first part identifies the error stack trace and the second part "uniquely" identifies the error?

Reply:
The uniqueness you are looking for is found in the time-stamp :-)
The error would have: time-stamp (when), `log_userid` (who), `log_oec` (what), and the `message`, which is produced with the "troubleshoting" function will provide the traceback and variables with (where, how)

Reply:
If the timestamp is propagated to the user I would be fine. But since this will not be the case I think it's still lacking.
I would like the best of both worlds. Uniqueness in the error code and a way to easily search for the specific error the user reported.

Reply:
On second thought, I will split the OEC into two: support_id (that is passed to the user and is like a "issue ticket number") and `error_fingerprint_code` (that is this).

Reply:
@GitHK New oec format captures both the traceback and an unique id `OEC:{traceback}-{timestamp}`